wtss_plugin/controller/files_export/files_export.py

```diff
@@ -51,6 +50,82 @@ def defaultCode(self):
         )
         return open(template, 'r').read()
 
+    def to_dataframe(self, time_series):
+        """Convert time series dict to dataframe."""
+        time_series_df = pandas.DataFrame({
+            "Index": [pandas.to_datetime(date) for date in time_series['result']['timeline']]
+        })
+        for result in time_series["result"]["attributes"]:
+            band = str(result["attribute"])
+            time_series_df[band] = result.get("values")
+        return time_series_df
+
+    def apply_to_time_series(
+            self, time_series: any,
+            type: str,
+            bands_description: any,
+            normalize_data: bool,
+            interpolate_data: bool
+        ):
+        """Apply normalize and interpolation to time series data."""
+        if type == "DataFrame":
+            for band in self.get_bands_from_df(time_series):
+                if normalize_data:
+                    def _normalize(value):
```

Comment:
You should not define a function inside a loop.
You may create an high level function which wrap up your closure variables and then annotate in `series.apply(myfn)`